### PR TITLE
Rafadana/event participants

### PR DESCRIPTION
**Pull request by RafadanaM**
### Background: 
Adding participants list page

### Additional Packages/Dependencies:
- none

### Changes:
- lib/app.dart
- lib/auth/data/user_model.dart
- lib/auth/data/user_model.g.dart
- lib/common/components/buttons/back_button.dart
- lib/common/components/buttons/custom_back_button.dart
- lib/common/config/route_generator.dart
- lib/common/config/theme.dart
- lib/event/bloc/event_detail_cubit.dart
- lib/event/components/participant_card.dart
- lib/event/data/event_detail_model.dart
- lib/event/data/event_detail_model.g.dart
- lib/event/data/event_location_model.dart
- lib/event/data/event_location_model.g.dart
- lib/event/data/event_participant_model.dart
- lib/event/data/event_participant_model.g.dart
- lib/event/data/participant_location_model.dart
- lib/event/data/participant_location_model.g.dart
- lib/page/edit_event.dart
- lib/page/edit_profile.dart
- lib/page/email_confirmation.dart
- lib/page/event_detail.dart
- lib/page/event_participants.dart
- lib/page/profile.dart
- lib/user/bloc/edit_user_cubit.dart
- lib/user/data/edit_user_data_provider.dart
- lib/user/data/edit_user_repository.dart
- lib/user/data/models/edit_user_model.dart
- lib/user/data/models/edit_user_model.g.dart
- lib/user/data/models/location_model.dart
- lib/user/data/models/location_model.g.dart

### How to Implement:
- none

### CODE REVIEW

File: lib/app.dart [hunk 1]
```diff
@@ -57,7 +58,10 @@ class _AppViewState extends State<AppView> {
           listener: (context, state) {
             if (state is AuthAuthenticatedState) {
               _navigator.pushNamedAndRemoveUntil(
-                  LocationPermission.routeName, (route) => false);
+                  // LocationPermission.routeName, (route) => false);
+                  EventDetail.routeName,
```
Comment (YafiAhsanH) [inline lib/app.dart]: you forgot to change this back to location permission

File: lib/common/components/buttons/back_button.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_boilerplate/common/config/theme.dart';
+
+class BackButton extends StatelessWidget {
+  final VoidCallback onPressesed;
```
Comment (YafiAhsanH) [inline lib/common/components/buttons/back_button.dart]: little typo, onPressed

File: lib/common/components/buttons/custom_back_button.dart [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_boilerplate/common/config/theme.dart';
+
+class CustomBackButton extends StatelessWidget {
+  final VoidCallback onPressed;
+  const CustomBackButton({Key? key, required this.onPressed}) : super(key: key);
+  @override
```
Comment (YafiAhsanH) [inline lib/common/components/buttons/custom_back_button.dart]: there are two back button components (BackButton and CustomBackButton) are they both being used?

File: lib/event/components/participant_card.dart [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  Widget build(BuildContext context) {
+    return Container(
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(CustomRadius.xxl),
+        boxShadow: [
+          BoxShadow(blurRadius: 4, color: neutral.shade300),
+        ],
+      ),
+      child: Card(
+        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20)),
```
Comment (YafiAhsanH) [inline lib/event/components/participant_card.dart]: border radius 20 can use the custom border radius

File: lib/page/email_confirmation.dart [hunk 5]
```diff
@@ -1,5 +1,8 @@
 import 'package:flutter/material.dart';
+
+import 'package:flutter_boilerplate/common/components/buttons/custom_back_button.dart';
 import 'package:flutter_boilerplate/common/config/theme.dart';
+
 import 'package:flutter_boilerplate/page/landing.dart';
```
Comment (YafiAhsanH) [inline lib/page/email_confirmation.dart]: u forgot to delete the empty lines

**Review by YafiAhsanH — CHANGES_REQUESTED** [review]

**Review by YafiAhsanH — APPROVED** [review]
looks good!

**Review by oliver-mf — APPROVED** [review]
Looks gud to me

**Pull request merged**